edge-modules/TwinTester/DesiredPropertyUpdater.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace TwinTester
+{
+    using System;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Shared;
+    using Microsoft.Extensions.Logging;
+
+    class DesiredPropertyUpdater : ITwinOperation
+    {
+        static readonly ILogger Logger = ModuleUtil.CreateLogger(nameof(DesiredPropertyUpdater));
+        readonly RegistryManager registryManager;
+        readonly ITwinTestResultHandler resultHandler;
```

Comment:
should this variable name as "reporter"?

Reply:
For alloperations mode (long haul) it actually saves to store and only reporting after validation, that's why I named it handler

---
In reply to: [359105066](https://github.com/Azure/iotedge/pull/2108#discussion_r359105066) [](ancestors = 359105066)